openshift/system/Dockerfile

```diff
@@ -1,91 +1,88 @@
-ADD lib/developer_portal/*.gemspec lib/developer_portal/
-ADD vendor/active-docs/*.gemspec vendor/active-docs/
-ADD Gemfile* ./
-ADD Rakefile ./
+ENV RUBY_MAJOR_VERSION=2 \
+    RUBY_MINOR_VERSION=7 \
+    RAILS_ENV=production \
+    SAFETY_ASSURED=1 \
+    TZ=:/etc/localtime
 
-COPY openshift/system/contrib/scl_enable /opt/app-root/etc/
+ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}"
 
-ENV BASH_ENV=/opt/app-root/etc/scl_enable \
-    ENV=/opt/app-root/etc/scl_enable \
-    PROMPT_COMMAND=". /opt/app-root/etc/scl_enable" \
-    RAILS_ENV=production \
-    SAFETY_ASSURED=1
+WORKDIR /opt/system
 
-RUN source /opt/app-root/etc/scl_enable \
-    && gem install bundler --version 2.2.25 \
-    && bundle config build.pg --with-pg-config=/usr/pgsql-13/bin/pg_config \
-    && bundle config set --local without development:test:licenses \
-    && bundle install --deployment --jobs $(grep -c processor /proc/cpuinfo) --retry=5
+RUN dnf -y module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
+    && yum install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info postgresql rubygem-irb rubygem-rdoc ruby libpq mysql mysql-libs gd git liberation-sans-fonts file \
+    && yum -y clean all
 
-RUN chgrp root /opt/system/
+COPY --from=builder /opt/system/ .
+COPY --from=builder /tmp/memkind-*/jemalloc/lib/libjemalloc.so* /usr/local/lib64
+COPY --from=builder /usr/local/share/gems /usr/local/share/
+COPY --from=builder /usr/local/bin/bundle* /usr/local/bin
+# in RHEL 8 podman-catatonit pulls in too many useless deps so we don't install the RPM directly
+COPY --from=builder /usr/libexec/catatonit/catatonit /usr/libexec/catatonit/catatonit
 
-ADD . ./
-ADD package.json ./
-ADD yarn.lock ./
-ADD openshift/system/config/* ./config/
+RUN echo /usr/local/lib64 > /etc/ld.so.conf.d/jemalloc.conf \
+    && ldconfig && ldconfig -p | grep jemalloc
 
-RUN source /opt/app-root/etc/scl_enable \
+RUN bundle config set --local without development:test:licenses:assets \
     && bundle exec rake tmp:create \
-    && mkdir -p public/assets db/sphinx \
+    && mkdir -p db/sphinx \
     && chmod g+w -vfR log tmp public/assets db/sphinx \
     && umask 0002 \
-    && cd /opt/system \
-    && npm install -g yarn \
-    && yarn install:safe --no-progress \
-    && SECRET_KEY_BASE=rails/32947 bundle exec rake assets:precompile tmp:clear \
-    && rm log/*.log \
+    && rm -f log/*.log \
     && chmod g+w /opt/system/config
 
-ADD openshift/system/entrypoint.sh /opt/system/entrypoint.sh
+# Bundler runs git commands on git dependencies
+# https://bundler.io/guides/git.html#local-git-repos
+# git will check if the current user is the owner of the git repository folder
+# This was included in git v2.35.2 or newer.
+# https://github.com/git/git/commit/8959555cee7ec045958f9b6dd62e541affb7e7d9
+# Openshift changes the effective UID, so this git check needs to be bypassed.
+RUN git config --global --add safe.directory '*'
```

Comment:
I need help to understand this. Are we using some gem from a local repository inside the container?

Reply:
We have gems from git and not from packaged gems. This is needed for them. This approach is already used by zync and backend.